pkg/commands/workdir.go

```diff
@@ -81,14 +84,116 @@ func (w *WorkdirCommand) ExecuteCommand(config *v1.Config, buildArgs *dockerfile
 
 // FilesToSnapshot returns the workingdir, which should have been created if it didn't already exist
 func (w *WorkdirCommand) FilesToSnapshot() []string {
-	return w.snapshotFiles
+	return nil
+}
+
+func (r *WorkdirCommand) ProvidesFilesToSnapshot() bool {
+	return false
 }
 
 // String returns some information about the command for the image config history
 func (w *WorkdirCommand) String() string {
 	return w.cmd.String()
 }
 
+// CacheCommand returns true since this command should be cached
+func (w *WorkdirCommand) CacheCommand(img v1.Image) DockerCommand {
+
+	return &CachingWorkdirCommand{
+		img:       img,
+		cmd:       w.cmd,
+		extractFn: util.ExtractFile,
+	}
+}
+
 func (w *WorkdirCommand) MetadataOnly() bool {
 	return false
```

Comment:
sometimes calling `WORKDIR` is metadata only, there might be some optimization potential here, depending on whether we can know this a-priori.

ie. calling 
```
WORKDIR /
```
without any further context is guaranteed to be metadata only in all images to my understanding.